pkg/spanconfig/spanconfigkvsubscriber/kvsubscriber.go

```diff
@@ -439,6 +440,27 @@ func (s *KVSubscriber) setLastUpdatedLocked(ts hlc.Timestamp) {
 func (s *KVSubscriber) handlePartialUpdate(
 	ctx context.Context, ts hlc.Timestamp, events []rangefeedbuffer.Event,
 ) {
+	// The events we've received from the rangefeed buffer are sorted in
+	// increasing timestamp order. However, any updates with the same timestamp
+	// may be ordered arbitrarily. That's okay if they don't overlap. However, if
+	// they do overlap, the assumption is that an overlapping delete should be
+	// ordered before an addition it overlaps with -- not doing would cause the
+	// addition to get clobbered by the deletion, which will result in the store
+	// having missing span configurations. As such, we re-sort the list of events
+	// before applying it to our store, using Deletion() as a tie-breaker when
+	// timestamps are equal.
+	sort.Slice(events, func(i, j int) bool {
+		switch events[i].Timestamp().Compare(events[j].Timestamp()) {
+		case -1: // ts(i) < ts(j)
+			return true
+		case 1: // ts(i) > ts(j)
+			return false
+		case 0: // ts(i) == ts(j); deletions sort before additions
+			return events[i].(*BufferEvent).Deletion() // no need to worry about the sort being stable
```

Comment:
Should we have added `Deletion() bool` to the `rangefeedbuffer.Event` interface?

Reply:
I wasn't suggesting that, but since you brought it up, should we do that?

To me (someone not particularly familiar with this package), it seems off for this `handlePartialUpdate` function to operate on `rangefeedbuffer.Event` interfaces, but then to assume that they are always `*BufferEvent` instances. I now see that we're also doing this later in the function though, so I guess this is nothing new. So we don't need to hold up this backport on this.

Maybe what I'm missing is that these `*BufferEvent` instances are coming from the same `spanconfigkvsubscriber` package, and are only passing through an interface before being handed back to `spanconfigkvsubscriber`. So what we really want is to parameterize our use of the `rangefeedcache` package with our `Event` type. Something like https://github.com/cockroachdb/cockroach/pull/119870.

Reply:
> I wasn't suggesting that, but since you brought it up, should we do that?

I considered this initially, but then chose not to change any of the other users of `rangefeedbuffer` in favour of keeping this change targeted. We could revisit this, but with your cleanup in #119870, I think it's fine. Thanks for sending that out btw! 

I'll merge this once CI is happy.